### DISCUSSION TITLE
Handle setting default values with api response having null values

### DISCUSSION THREAD

**Discussion by ramseylove** [react-hook-form/react-hook-form#10240] [Q&A]
The api i'm using sends a null value in a field if there is no data for the field. A few fields requires a textArea component that does not like the null value.  I get this error:
`'value' prop on 'textarea' should not be null. Consider using an empty string to clear the component or 'undefined' for uncontrolled components.`

I'm setting the defaultValues to my api response values. I do not want to change the value unless user actually changes the values (keep null if still blank).   Right now i only submit "dirtyFields"  and if i set to blank string, it will submit those fields with empty strings. Does anyone have an idea how I can handle this case?

I do not really want to change the value if user doesn't input anything because it will overwrite the null value on the server. but I tried doing a setValue on the field if value is null and got another error:
`cannot update a component ('ForwardRef(RhfControlledInput)') while rendering a different component ('Controller'). To locate the bad setState() call inside 'Controller', ` 

```tsx
<Controller
          name="requestHeaders"
          control={form.control}
          render={({ field }) => {
            // if (field.value === null) form.setValue('requestHeaders', ''); // causes setState error
            return (
              <TextAreaWithLabels
                {...field}
                rows={6}
                value={field.value}
                onChange={(e) => field.onChange(e.value)}
                label={'Headers'}
              />
            );
          }}
        />
```

**Comment by fahadsohail482** [comment]
@ramseylove One way to do this is to modify the response from api to check weather it has null value if yes then replace it with an empty string or whatever you like to set and then use reset to set values to form.
For example.
`const data=await getData() // [{email:"ok@example.com", foo : null}]`
`const modifiedData = data.map((item)=>({...item, foo: item.foo ?? ""}))`
`//reset(modifiedData)`
`//setState(modifiedData)`
your thoughts @ramseylove ?

**Reply by mk48** [reply]
Setting empty string for text field is OK.
any suggestion for Number field?
I have the same issue with nullable number field in DB.
